crates/forge/src/running.rs

```diff
@@ -67,18 +83,55 @@ pub(crate) fn run_from_test_case(
         footer.iter()
     );
     let (hints_dict, string_to_hint) = build_hints_dict(instructions.clone());
-    let core_cairo_hint_processor = CoreCairoHintProcessor {
-        runner: Some(runner),
-        starknet_state: StarknetState::default(),
-        string_to_hint,
-        run_resources: RunResources::default(),
+
+    // Losely inspired by crates/cheatnet/src/execution/cairo1_execution::execute_entry_point_call_cairo1
+    let block_context = build_block_context();
+    let account_context = build_transaction_context();
+    let mut context = EntryPointExecutionContext::new(
+        block_context.clone(),
+        account_context,
+        block_context.invoke_tx_max_n_steps.try_into().unwrap(),
+    );
+    let test_selector = get_selector_from_name("TEST_CONTRACT_SELECTOR").unwrap();
+    let entry_point_selector = EntryPointSelector(StarkHash::new(test_selector.to_bytes_be())?);
+    let entry_point = CallEntryPoint {
+        class_hash: None,
+        code_address: Some(ContractAddress::from(0_u8)),
+        entry_point_type: EntryPointType::External,
+        entry_point_selector,
+        calldata: Calldata(Arc::new(vec![])),
+        storage_address: ContractAddress(patricia_key!("0x0")),
+        caller_address: ContractAddress::default(),
+        call_type: CallType::Call,
+        initial_gas: u64::MAX,
     };
+
+    let mut blockifier_state = CachedState::new(
+        build_testing_state(predeployed_contracts),
+        GlobalContractCache::default(),
+    );
+    let mut execution_resources = ExecutionResources::default();
+    let syscall_handler = SyscallHintProcessor::new(
+        &mut blockifier_state,
+        &mut execution_resources,
+        &mut context,
+        // This segment is created by SierraCasmRunner
+        Relocatable {
+            segment_index: 10,
+            offset: 0,
+        },
+        entry_point,
+        &string_to_hint,
+        ReadOnlySegments::default(),
+    );
     let mut cairo_hint_processor = CairoHintProcessor {
-        original_cairo_hint_processor: core_cairo_hint_processor,
+        blockifier_syscall_handler: syscall_handler,
         contracts,
         cheatnet_state: CheatnetState::new(StateReaderProxy(Box::new(build_testing_state(
             predeployed_contracts,
         )))),
+        hints: &string_to_hint,
+        run_resources: RunResources::new(0),
```

Comment:
We used `RunResources::default()` in `CoreCairoHintProcessor` before (which returns `None`). Why do we use 0 here instead?

Reply:
I think default is 0, but I will replace it to default for consistency